From 0b4b5982c5483121797f59dc9139dfe5dc7c7da5 Mon Sep 17 00:00:00 2001
Message-Id: <0b4b5982c5483121797f59dc9139dfe5dc7c7da5.1421886431.git.chang-joon.lee@intel.com>
In-Reply-To: <2a59efb195329ac28200fa6e68e1170a4cf8d03c.1421886431.git.chang-joon.lee@intel.com>
References: <2a59efb195329ac28200fa6e68e1170a4cf8d03c.1421886431.git.chang-joon.lee@intel.com>
From: Durgadoss R <durgadoss.r@intel.com>
Date: Fri, 19 Dec 2014 10:49:12 +0530
Subject: [PATCH 03/11] FROM_UPSTREAM [VPG]: drm/i915: Introduce
 intel_fb_obj() macro

Add an intel_fb_obj() macro that returns the GEM object associated with
a DRM framebuffer.  This macro is safe to call on NULL framebuffers (a
NULL object pointer will be returned in this case).

Signed-off-by: Matt Roper <matthew.d.roper@intel.com>
Reviewed-by: Chris Wilson <chris@chris-wilson.co.uk>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>

Change-Id: If6e8eb670617ef0bf4ce818606cc4719d58f5f84
Upstream-repo: git://anongit.freedesktop.org/drm-intel
Upstream-commit: 155e63696caab748da3f344f66de503038a72024
Issue: GMINL-4897
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 drivers/gpu/drm/i915/intel_drv.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/i915/intel_drv.h b/drivers/gpu/drm/i915/intel_drv.h
index b311780..29a2efa 100644
--- a/drivers/gpu/drm/i915/intel_drv.h
+++ b/drivers/gpu/drm/i915/intel_drv.h
@@ -599,6 +599,7 @@ struct cxsr_latency {
 #define to_intel_encoder(x) container_of(x, struct intel_encoder, base)
 #define to_intel_framebuffer(x) container_of(x, struct intel_framebuffer, base)
 #define to_intel_plane(x) container_of(x, struct intel_plane, base)
+#define intel_fb_obj(x) (x ? to_intel_framebuffer(x)->obj : NULL)
 
 /* HDMI bits are shared with the DP bits */
 #define   HDMIB_HOTPLUG_LIVE_STATUS             (1 << 29)
-- 
1.7.9.5

